### PR TITLE
Qt/NetPlayBrowser: Various small fixes

### PR DESCRIPTION
**Pull request by spycrab**
Changes:
* Fixes the In-Game column being editable
* Fixes the header getting bold when an item is selected
* Adds a version column
* Adds an option to show incompatible sessions

### CODE REVIEW

File: Source/Core/DolphinQt/NetPlay/NetPlayBrowser.cpp [hunk 1]
```diff
@@ -67,6 +68,9 @@ void NetPlayBrowser::CreateWidgets()
   m_button_refresh = new QPushButton(tr("Refresh"));
   m_edit_name = new QLineEdit;
   m_edit_game_id = new QLineEdit;
+  m_check_hide_incompatible = new QCheckBox(tr("Hide incompatible sessions"));
```
Comment (Techjar) [inline Source/Core/DolphinQt/NetPlay/NetPlayBrowser.cpp]: Should be Title Capitalized for consistency with most other UI things.

**Comment by Techjar** [comment]
I have a couple suggestions to improve usability:
1. Make incompatible versions red or something to make it obvious at a glance.
2. Disable hide incompatible sessions by default, as I think it's likely to just confuse users when they don't see their friend's session.

**Comment by spycrab** [comment]
> Disable hide incompatible sessions by default, as I think it's likely to just confuse users when they don't see their friend's session.

No, I actually think that is more confusing as it presents users with a big list of entries that they will not be able to interact with. 

The default UI state should be the most useful one, so I'd keep it this way.

> Make incompatible versions red or something to make it obvious at a glance.

It being disabled should be obvious enough. Marking items as red also has many caveats regarding themability and accessibility.

**Comment by Techjar** [comment]
> No, I actually think that is more confusing as it presents users with a big list of entries that they will not be able to interact with.
I guess there's potential problems on both sides, but enabled by default presents a nicer UI, so fair enough.

> It being disabled should be obvious enough. Marking items as red also has many caveats regarding themability and accessibility.
True. And I guess if hide incompatible is on by default, then it should be obvious that disabled means incompatible once a user turns that option off.

While we're fixing issues with this UI, could you also either remove the list refresh from onTextChanged or do the refresh asynchronously? My UI hangs for sometimes tens of seconds when typing into the Name and Game ID fields.

I also had the idea to have the disabled rows link to the appropriate build download page (through dolp.in), if it's feasible to do this. What are your thoughts on this?

**Comment by spycrab** [comment]
I do not want to extend the scope of this PR beyond what's already in it

**Pull request merged**